src/irmin-layers/irmin_layers_intf.ml

```diff
@@ -28,6 +28,38 @@ module type S = sig
     ?recovery:bool ->
     repo ->
     unit Lwt.t
+  (** [freeze ?min ?max ?squash ?copy_in_upper ?min_upper ?recovery t] launches
+      an asynchronous freezing operation on the repo [t] to reduce the size of
+      the upper layer.
+
+      Let [o] be the set of objects (i.e. commits, nodes and contents) reachable
+      from the [max] commits and bounded by the [min] commits. During the
+      freeze, all objects in [o] are copied to the lower layer, if there is one.
+      Setting [squash] to [true] is equivalent to setting [min] to [max]. [min]
+      defaults to the empty list and [max] defaults to the head commits of the
+      repo.
+
+      Let [o'] be the set of objects reachable from the [max] commits and
+      bounded by the [min_upper] commits. When the freeze is over, if
+      [copy_in_upper] is true, then the upper layer will only contain the
+      objects of [o'], otherwise the upper layer will left be empty. [min_upper]
+      defaults to the empty list and [copy_in_upper] defaults to the repo's
+      configuration.
+
+      If [recovery] is true then the function will first try to recover from a
+      previously interrupted freeze. See {!needs_recovery}.
+
+      If a freeze is already ongoing, the behavior depends on the
+      freeze_throttle configuration of the repo:
+
+      - When [`Overcommit_memory], the function returns without launching a new
+        freeze.
```

Comment:
Given that this doesn't really have an impact on the memory behavior of the freeze (as a few objects will have to be kept in memory anyway as the previous freeze is still running) maybe this option should be called differently. Maybe `Cancel_new` ?

Reply:
All 3 tags could use a renaming IMO.

I like `Cancel_new` alongside `Cancel_current`.

We could also do `` `Block | `Ignore | `Supersede``

Reply:
This looks good to me.